### PR TITLE
DBZ-1661 Verify streaming off of secondary works

### PR DESCRIPTION
**Pull request by MartinMedek**
…g from mysql slave

### CODE REVIEW

**Review by obabec — CHANGES_REQUESTED** [review]

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/ConfigProperties.java [hunk 1]
```diff
@@ -69,6 +71,7 @@ private ConfigProperties() {
     public static final String DATABASE_MYSQL_PASSWORD = System.getProperty("test.database.mysql.password", "mysqlpw");
     public static final String DATABASE_MYSQL_DBZ_USERNAME = System.getProperty("test.database.mysql.dbz.username", "debezium");
     public static final String DATABASE_MYSQL_DBZ_PASSWORD = System.getProperty("test.database.mysql.dbz.password", "dbz");
+
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/ConfigProperties.java:74]: Unnecessary new lines in this file

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mysql/DockerMySqlDeployer.java [hunk 2]
```diff
@@ -5,7 +5,7 @@
  */
 package io.debezium.testing.system.tools.databases.mysql;
 
-import static io.debezium.testing.system.tools.ConfigProperties.DOCKER_IMAGE_MYSQL;
+import static io.debezium.testing.system.tools.ConfigProperties.DOCKER_IMAGE_MYSQL_MASTER;
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mysql/DockerMySqlDeployer.java:8]: I think we probably can avoid this naming change, adding MASTER suffix is not necessary. Out of the box if you deploy a single replica you need a master anyway. So I would keep just DOCKER_IMAGE_MYSQL and add just DOCKER_IMAGE_MYSQL_REPLICA.

File: debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mysql/OcpMySqlMasterDeployer.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                                   PersistentVolumeClaim volumeClaim,
+                                   OpenShiftClient ocp) {
+        super(project, deployment, services, pullSecret, ocp);
+        this.volumeClaim = volumeClaim;
+    }
+
+    @Override
+    public MySqlMasterController deploy() {
+        LOGGER.info("Deploying persistent volume claim");
+        ocp.persistentVolumeClaims().inNamespace(ConfigProperties.OCP_PROJECT_MYSQL).createOrReplace(volumeClaim);
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/mysql/OcpMySqlMasterDeployer.java:44]: Strange formatting. Or at least it looks strange in GH

File: debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/tests/mysql/MySqlTests.java [hunk 4]
```diff
@@ -37,22 +38,36 @@ public MySqlTests(
         super(kafkaController, connectController, connectorConfig, assertions);
     }
 
-    public void insertCustomer(
-                               SqlDatabaseController dbController,
-                               String firstName, String lastName,
-                               String email)
+    public static void insertCustomer(
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/tests/mysql/MySqlTests.java:41]: I don't see any reason for static context here. Can you explain?

**Review by obabec — CHANGES_REQUESTED** [review]

File: debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/tests/mysql/MySqlTests.java (hunk 4 above)
Comment (MartinMedek) [reply debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/tests/mysql/MySqlTests.java:41]: This method is used also in the OcpMysqlReplicationIT, I also considered moving this method to another helper class, but I thought it's unnecessary .

File: debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/tests/mysql/MysqlTestsWithDbOperations.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+import io.debezium.testing.system.assertions.KafkaAssertions;
+import io.debezium.testing.system.tests.ConnectorTest;
+import io.debezium.testing.system.tools.databases.SqlDatabaseClient;
+import io.debezium.testing.system.tools.databases.SqlDatabaseController;
+import io.debezium.testing.system.tools.kafka.ConnectorConfigBuilder;
+import io.debezium.testing.system.tools.kafka.KafkaConnectController;
+import io.debezium.testing.system.tools.kafka.KafkaController;
+
+public class MysqlTestsWithDbOperations extends ConnectorTest {
```
Comment (obabec) [inline debezium-testing/debezium-testing-system/src/test/java/io/debezium/testing/system/tests/mysql/MysqlTestsWithDbOperations.java:21]: This class is not necessary. And also you should take more time thinking about naming. You created a class that is named Tests but there are no tests :) It is simply a utility class we should avoid as much as possible...

**Review by obabec — CHANGES_REQUESTED** [review]

**Comment by jpechane** [comment]
@MartinMedek It seems there is a conflit in the release pipeline. Did you pull the latest main?

**Comment by MartinMedek** [comment]
> @MartinMedek It seems there is a conflit in the release pipeline. Did you pull the latest main?

should be fixed now

**Comment by jpechane** [comment]
@obabec Is there anything else we are waiting for?

**Comment by obabec** [comment]
@jpechane nope, we can merge :)

**Pull request merged**